src/Refactoring-UI/ReExtractToTemporaryDriver.class.st

```diff
@@ -78,7 +80,10 @@ ReExtractToTemporaryDriver >> runRefactoring [
 	refactoring prepareForExecution.
 	
 	newVariableName := self requestVariableName.
+	refactoring extract: sourceInterval to: newVariableName from: selector in: class.
```

Comment:
Instead of calling the constructor again, here you can just create setter in the refactoring that sets the new name